scripts/release/upload_metadata_file.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+import os
+import shutil
+import subprocess
+import zipfile
+from functools import cache
+from pathlib import Path
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Upload metadata file to S3")
+    parser.add_argument(
+        "--package", type=str, required=True, help="Path to the package"
+    )
+    parser.add_argument(
+        "--bucket", type=str, required=True, help="S3 bucket to upload metadata file to"
+    )
+    parser.add_argument(
+        "--key-prefix",
+        type=str,
+        required=True,
+        help="S3 key to upload metadata file to",
+    )
+    parser.add_argument("--dry-run", action="store_true", help="Dry run")
+    args = parser.parse_args()
+    if args.bucket.startswith("s3://"):
+        args.bucket = args.bucket[5:]
+    if args.key_prefix.startswith("/"):
+        args.key_prefix = args.key_prefix[1:]
+    if args.key_prefix.endswith("/"):
+        args.key_prefix = args.key_prefix[:-1]
```

Comment:
I'm a bit unsure what these lines are doing, might be good to leave a comment on why these are necessary

Reply:
It's sanitizing the args a bit, added a comment saying this